src/hybrid_reader.c

```diff
@@ -312,7 +341,8 @@ void HybridIterator_Free(struct indexIterator *self) {
   if (it == NULL) {
     return;
   }
-  if (it->searchMode == VECSIM_HYBRID_ADHOC_BF || it->searchMode == VECSIM_HYBRID_BATCHES) {
+  if (it->searchMode == VECSIM_HYBRID_ADHOC_BF || it->searchMode == VECSIM_HYBRID_BATCHES ||
+      it->searchMode == VECSIM_HYBRID_BATCHES_TO_ADHOC_BF) {
```

Comment:
change to `if (it->topResults)` and `if (it->returnedResults)`

Reply:
This is not equivalent... `it->topResults` and `it->returnedResults` are not initialized to NULL, therefore they may contain garbage. That can be changed of course...